controllers/istiorevision/istiorevision_controller.go

```diff
@@ -226,28 +226,26 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 		// namespaced resources
 		Watches(&corev1.ConfigMap{}, ownedResourceHandler).
-		Watches(&appsv1.Deployment{}, ownedResourceHandler).
-		Watches(&appsv1.DaemonSet{}, ownedResourceHandler).
-		Watches(&corev1.Endpoints{}, ownedResourceHandler).
-		Watches(&corev1.ResourceQuota{}, ownedResourceHandler).
+		Watches(&appsv1.Deployment{}, ownedResourceHandler). // we don't ignore the status here because we use it to calculate the IstioRevision status
+		Watches(&appsv1.DaemonSet{}, ownedResourceHandler).  // we don't ignore the status here because we use it to calculate the IstioRevision status
+		Watches(&corev1.Endpoints{}, ownedResourceHandler, builder.WithPredicates(ignoreStatusChange())).
```

Comment:
Just curious: Can you please explain why we are watching Endpoints? 
Also, should we have a watch on Endpointslices as well?

Reply:
The base chart creates the Endpoints directly: https://github.com/istio-ecosystem/sail-operator/blob/fd8c600b163cb0554688cb03e95442e5786694ab/resources/latest/charts/base/templates/endpoints.yaml

No need to watch Endpointslices, because we don't create them directly and thus they won't have an ownerReference pointing to the IstioRevision. So, even if we watched them, we'd ignore them because they wouldn't have the ownerReference.